docs/proposals/08-shoot-additional-container-runtimes.md

```diff
@@ -0,0 +1,139 @@
+        3. os-suse-jeos - __TBD__     
+        
+    3. Docker-monitor daemon and rotate log should be replaced with equivalent conatinerd services. __TBD__ 
+
+2. Validate workers additional runtime configurations.  
+3. Add support for each additional container runtime in the cluster.   
+    1. In order to install each additional available runtime in the cluster we should:
+        1. Install the runtime binaries in each Worker's pool nodes that specified the runtime support.
+        2. Apply the relevant RuntimeClass to the cluster.
+        
+    2. The installation above should be done by a new kind of extension: RuntimeContainer resource. For each runtime container type (Kata-container/gvisor) a dedicate extension controller will be created. 
+        1. A label for each container runtime support will be added to every node that belongs to the worker pool. This should be done similar
+           to the way labels created today for each node, through kubelet execution parameters (_kubelet.flags: --node-labels). When creating the OperatingSystemConfig (original) for the worker each container runtime support should be mapped to a label on the node.
+           For Example:
+           label: container.runtime.kata-containers=true (shoot.spec.cloud.<IAAS>.worker.containerRuntimes.kata-container)
+           label: container.runtime.gvisor=true (shoot.spec.cloud.<IAAS>.worker.containerRuntimes.gvisor)
+        2. During the Shoot reconciliation (Similar steps to the Extensions today) Gardener will create new RuntimeContainer resource if a container runtime exist in at least one worker spec:
+            ```yaml
+            apiVersion: runtimecontainer.gardener.cloud/v1alpha1
+            kind: RuntimeContainer
+            metadata:
+              name: kata-containers-runtime-extention
+              namespace: shoot--foo--bar
+            spec:
+              type: kata-containers
+            ```   
+   
+            Gardener will wait that all RuntimeContainers extensions will be reconciled by the appropriate extensions controllers.
+        3. Each runtime extension controller will be responsible to reconcile it's RuntimeContainer resource type.
+           rc-kata-containers extension controller will reconcile RuntimeContainer resource from type kata-container and rc-gvisor will reconcile RuntimeContainer resource from gvisor.
+           Reconciliation process by container runtime extension controllers:
+           1. Runtime extension controller from specific type should apply a chart which responsible for the installation of the runtime container in the cluster:
+                1. DaemonSet which will run a privileged pod on each node with the label: container.runtime.<type of the resource>:true The pod will be responsible for:
+                    1. Download the runtime container binaries (From the web or from the extension package   __TBD__) and copy them to the relevant path in the host OS.
```

Comment:
Since this is simply a `DaemonSet`, the container that it copies should have the binaries built into it. No need to download from the web of from the extension controller.

Reply:
Again, see above, we should try to only download container images and copy necessary binaries out of it. cc @ccwienk @vlerenc

Reply:
Updated